examples/customresources/apis/thingamabobapi/thingamabobapi.go

```diff
@@ -0,0 +1,42 @@
+// Package thingamabobapi defines an empty API to demonstrate custom validation logic.
+package thingamabobapi
+
+import (
+	"context"
+
+	"github.com/edaniels/golog"
+	testecho "go.viam.com/api/component/testecho/v1"
+	"go.viam.com/rdk/registry"
+	"go.viam.com/rdk/resource"
+	"go.viam.com/rdk/subtype"
+	echopb "go.viam.com/utils/proto/rpc/examples/echo/v1"
+	"go.viam.com/utils/rpc"
+	echoserver "go.viam.com/utils/rpc/examples/echo/server"
+)
+
+var Subtype = resource.NewSubtype(
+	resource.Namespace("acme"),
+	resource.ResourceTypeComponent,
+	resource.SubtypeName("thingamabob"),
+)
+
+type Thingamabob interface{}
+
+func init() {
+	// Register a simple echo service as a placeholder API.
+	registry.RegisterResourceSubtype(Subtype, registry.ResourceSubtype{
```

Comment:
Had very little idea what I was doing here. It seems like I needed to register the resource subtype, and I wasn't sure what service to provide.